src/GameLogic/Player.cs

```diff
@@ -1685,9 +1707,37 @@ private async ValueTask AfterKilledPlayerAsync()
         return null;
     }
 
+    private void RaisePlayerEnteredMap(GameMap map)
+    {
+        this.PlayerEnteredMap?.Invoke(this, (this, map));
+        if (map.Definition.CharacterPowerUpDefinitions is { Count: > 0 } powerUpDefinitions
+            && this.Attributes is { } attributes)
+        {
+            foreach (var powerUpDefinition in powerUpDefinitions)
+            {
+                if (powerUpDefinition.TargetAttribute is not { } targetAttribute)
+                {
+                    continue;
+                }
+
+                var powerUps = PowerUpWrapper.CreateByPowerUpDefinition(powerUpDefinition, attributes);
+                powerUps.ForEach(p =>
+                {
+                    this.Attributes?.AddElement(p, targetAttribute);
+                    this.PlayerLeftMap += (_, _) => attributes.RemoveElement(p, targetAttribute);
```

Comment:
Is it necessary to unsubscribe?

Reply:
yes, of course 😅

Reply:
@sven-n I mean PlayerLeftMap event. Is it automatically GC? AFAIK we need to manually unsubscribe from it like `this.PlayerLeftMap -= someHandler`.

Reply:
oops, I thought i commited that change...

Reply:
Another place without unsubscribe https://github.com/MUnique/OpenMU/blob/master/src/GameLogic/MiniGames/MiniGameContext.cs#L170C9-L170C69

Reply:
Thanks, fixed with #373